### ISSUE TITLE
Patching pointer values

### ISSUE THREAD

**Issue by stoffen** [r3labs/diff#72]
Hi, In the following I am unable to patch the struct. To me it seems that the diff holds a string and that the reflection machinery does not consider the target pointer. I am missing something here? Thanks:)

```
type Test struct {
	S *string `json:"s,omitempty"`
}

func TestPointerDiff(t *testing.T) {
	str1 := "before"
	t1 := Test{S: &str1}
	str2 := "after"
	t2 := Test{S: &str2}

	changelog, err := diff.Diff(t1, t2)
	assert.NoError(t, err)

	patchLog := diff.Patch(changelog, t1)
	assert.False(t, patchLog.HasErrors())
}
```

**Comment by purehyperbole** [comment]
Hey @stoffen ,

Thanks for raising the issue! Looks like it was an issue calling `reflect.Set()`. This should now be fixed with `v2.14.3`.

Please note, when calling patch, you will need to pass in a pointer to the value you're patching to, so your example would become: 

```go
	// pass in t1 as pointer
	patchLog := diff.Patch(changelog, &t1)
	assert.False(t, patchLog.HasErrors())
```

**Comment by stoffen** [comment]
Thanks, you just saved my day
